rapid/src/main/java/com/vrg/rapid/MembershipService.java

```diff
@@ -338,6 +351,35 @@
         return future;
     }
 
+    /**
+     * Compiles a list of view changes that occured since the requesting node last saw a change
+     */
+    private ListenableFuture<RapidResponse> handleMessage(final SynchronizationMessage request) {
+        final SettableFuture<RapidResponse> future = SettableFuture.create();
+
+        // retrieve all the configuration changes after the one of the request
+        final List<ViewChange> changeList = new ArrayList<>();
+
+        boolean configurationIdReached = false;
+        for (final ViewChange viewChange : membershipHistory) {
```

Comment:
Same comment as elsewhere: would be faster to scan from the most recent change to the oldest change.